### DISCUSSION TITLE
Can PySR be used to search for equivalent circuit models?

### DISCUSSION THREAD

**Discussion by ma-sadeghi** [MilesCranmer/PySR#542] [Q&A]
Thanks for creating this amazing package!

Following #540 (and your recent Linkedin post), I was wondering if PySR can be used to search the space of possible equivalent circuit models given impedance-frequency measurements.

Just to be clear, I'm not trying to abuse your kindness in responding to such broad questions and even going above and beyond to provide a fully working prototype! (as you've kindly done in #540) I just want to know if PySR is the right tool for such problems, and possibly some high level pointers on where to start.

Thanks once again!

**Comment by MilesCranmer** [comment]
Hi @ma-sadeghi,

Thanks for the question! The current requirement is that all operators need to have a scalar input (1 for unary; 2 for binary) and scalar output. If you can work with that then there is probably a way to write a loss function for it. For other types of inputs; SymbolicRegression.jl will need to be extended (help always appreciated!)
